newIDE/app/src/GameDashboard/GamesList.js

```diff
@@ -44,19 +52,82 @@ export const GamesList = ({
         setError(null);
         const games = await getGames(getAuthorizationHeader, firebaseUser.uid);
         setGames(games);
-        // If a game id was passed, open it.
-        if (initialGameId) {
+      } catch (error) {
+        console.error('Error while loading user games.', error);
+        setError(error);
+      }
+    },
+    [authenticated, firebaseUser, getAuthorizationHeader]
+  );
+
+  const onRegisterGame = React.useCallback(
+    async () => {
+      if (!profile || !project) return;
+
+      const { id } = profile;
+      try {
+        setIsGameRegistering(true);
+        await registerGame(getAuthorizationHeader, id, {
+          gameId: project.getProjectUuid(),
+          authorName: project.getAuthor() || 'Unspecified publisher',
+          gameName: project.getName() || 'Untitled game',
+          templateSlug: project.getTemplateSlug(),
+        });
+        await loadGames();
+      } catch (error) {
+        console.error('Unable to register the game', error);
+        if (error.response && error.response.status === 403) {
+          await showAlert({
+            title: t`Game already registered`,
+            message: t`The project currently opened is registered online but you don't have
+          access to it. Ask the original owner of the game to share it with you
+          to be able to manage it.`,
+          });
+        } else {
+          await showAlert({
+            title: t`Unable to register the game`,
+            message: t`An error happened while registering the game. Verify your internet connection
+          or retry later.`,
+          });
+        }
+      } finally {
+        setIsGameRegistering(false);
+      }
+    },
+    [getAuthorizationHeader, profile, project, showAlert, loadGames]
+  );
+
+  React.useEffect(
+    () => {
+      const loadInitialGame = async () => {
+        // When games are loaded and we have an initial game id, open it.
+        if (games && initialGameId) {
           const game = games.find(game => game.id === initialGameId);
           if (game) {
             setOpenedGame(game);
+          } else {
+            onGameDetailsDialogClose(); // Ensure we reset initial props.
```

Comment:
Looks a bit like a hack? Add a longer comment because it's not clear why asking for the close of the game details dialog will "reset initial prop" for the reader.